codec-http2/src/main/java/io/netty/handler/codec/http2/CompressorHttp2ConnectionEncoder.java

```diff
@@ -147,17 +147,43 @@ public ChannelFuture writeData(final ChannelHandlerContext ctx, final int stream
     @Override
     public ChannelFuture writeHeaders(ChannelHandlerContext ctx, int streamId, Http2Headers headers, int padding,
             boolean endStream, ChannelPromise promise) {
-        initCompressor(streamId, headers, endStream);
-        return super.writeHeaders(ctx, streamId, headers, padding, endStream, promise);
+        try {
```

Comment:
Yes there are changes `CompressorHttp2ConnectionEncoder` is broken. The compressor was broken and discovered while do the re-work.  I could break it up into another PR...